src/internal/base-component/validate-props.ts

```diff
@@ -0,0 +1,26 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { isDevelopment } from '../is-development';
+
+export function validateProps(
+  componentName: string,
+  props: Record<string, any>,
+  excludedProps: Array<string>,
+  allowedEnums: Record<string, Array<string>>,
+  systemName = 'default'
```

Comment:
I think this property should be required, we can control when the method is called so we should always be able to provide a meaningful value

Reply:
sounds good